packages/jest-core/src/SearchSource.ts

```diff
@@ -40,8 +40,12 @@ export type TestSelectionConfig = {
 const globsToMatcher = (globs: Array<Config.Glob>) => (path: Config.Path) =>
   micromatch([replacePathSepForGlob(path)], globs, {dot: true}).length > 0;
 
-const regexToMatcher = (testRegex: Array<string>) => (path: Config.Path) =>
-  testRegex.some(testRegex => new RegExp(testRegex).test(path));
+const regexToMatcher = (testRegex: Array<string> | Array<RegExp>) => (
+  path: Config.Path,
+) =>
+  testRegex.some((testRegex: string | RegExp) =>
+    new RegExp(testRegex).test(path),
+  );
```

Comment:
```suggestion
const regexToMatcher = (testRegex: Config['testRegex']) => (
  path: Config.Path,
) =>
  testRegex.some(testRegex =>
    new RegExp(testRegex).test(path),
  );
```

maybe?

Reply:
![image](https://user-images.githubusercontent.com/27772165/78779180-c8dce400-799c-11ea-886c-c67b36371dda.png)
:(

Reply:
```
type TestRegex = Array<string | RegExp>;

const regexToMatcher = (testRegex: TestRegex) => (path: Config.Path) =>
  testRegex.some(testRegex => new RegExp(testRegex).test(path));
```
this works !

Reply:
`Config.ProjectConfig['testRegex']`

Reply:
done, beautiful now :)